btree/btree.c

```diff
@@ -4327,13 +4345,20 @@ int64_t btree_iter_kv_tick(struct m0_sm_op *smop)
 			 */
 			if (!node_isvalid(lev->l_node) ||
 			    !node_verify(lev->l_node)) {
-				m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_SETUP);
+				return m0_sm_op_sub(&bop->bo_op, P_CLEANUP,
+						    P_SETUP);
 			}
 
 			oi->i_key_found = node_find(&s, &bop->bo_rec.r_key);
 			lev->l_idx = s.s_idx;
 
 			if (node_level(s.s_node) > 0) {
+				if (oi->i_used == oi->i_height - 1) {
```

Comment:
similar check is needed at P_SIBLING for `i_pivot`.

Reply:
done.